diff --git a/react-native-image-resizer.podspec b/react-native-image-resizer.podspec
index 1234567..abcdef0 100644
--- a/react-native-image-resizer.podspec
+++ b/react-native-image-resizer.podspec
@@ -20,7 +20,9 @@ Pod::Spec.new do |s|
   s.source_files = "ios/**/*.{h,m,mm}"
   s.platform = :ios, "11.0"
 
-  s.dependency "RCT-Folly"
+  if ENV['RCT_USE_PREBUILT_RNCORE'] != '1' then
+    s.dependency "RCT-Folly"
+  end
   s.dependency "React-Core"
 
 end