diff --git a/react-native-quick-crypto.podspec b/react-native-quick-crypto.podspec
index 31a98d92abcbd52631aecd323a0ae7cb29c7e8c1..b880870391e1da329dad2f2bfcb29224ba947110 100644
--- a/react-native-quick-crypto.podspec
+++ b/react-native-quick-crypto.podspec
@@ -41,4 +41,5 @@ Pod::Spec.new do |s|
   s.dependency "React-Core"
   s.dependency "React"
   s.dependency "React-callinvoker"
+  s.dependency "ReactCommon"
 end
