diff --git a/android/build.gradle b/android/build.gradle
index 3ec9ff0bdaac411d7f96593ed30aa5799296197a..20110bbf78070dc24ba6d4423e15972552cd1f1c 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -118,7 +118,7 @@ android {
   }
 
   packagingOptions {
-    excludes = ["**/libc++_shared.so", "**/libfbjni.so", "**/libreactnativejni.so", "**/libjsi.so", "**/MANIFEST.MF", ""]
+    excludes = ["**/libc++_shared.so", "**/libfbjni.so", "**/libreactnativejni.so", "**/libjsi.so", "**/libreact_nativemodule_core.so", "**/libturbomodulejsijni.so", "**/MANIFEST.MF", ""]
     doNotStrip '**/*.so'
   }
 
