diff --git a/android/src/main/java/com/killserver/screenshotprev/RNScreenshotPreventModule.java b/android/src/main/java/com/killserver/screenshotprev/RNScreenshotPreventModule.java
index d0187a37fae44d0cf0b51616a884fb508f1c0e23..1589628f29f662820ce0b6706b5019d35fe52e15 100644
--- a/android/src/main/java/com/killserver/screenshotprev/RNScreenshotPreventModule.java
+++ b/android/src/main/java/com/killserver/screenshotprev/RNScreenshotPreventModule.java
@@ -26,51 +26,53 @@ public class RNScreenshotPreventModule extends ReactContextBaseJavaModule {
   @ReactMethod
   public void enabled(boolean _enable) {
     if (this.reactContext.hasCurrentActivity()) {
-      final Activity activity = this.reactContext.getCurrentActivity();
+      final Activity activity = this.activity;
       if (activity != null) {
         if (_enable) {
           activity.runOnUiThread(new Runnable() {
             @Override
             public void run() {
-              reactContext.getCurrentActivity().getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE);
+              activity.getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE,
+                  WindowManager.LayoutParams.FLAG_SECURE);
             }
           });
         } else {
           activity.runOnUiThread(new Runnable() {
             @Override
             public void run() {
-              reactContext.getCurrentActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
+              activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
             }
           });
         }
       }
     }
   }
-  
+
   @ReactMethod
   public void enableSecureView() {
     if (this.reactContext.hasCurrentActivity()) {
-      final Activity activity = this.reactContext.getCurrentActivity();
+      final Activity activity = this.activity;
       if (activity != null) {
         activity.runOnUiThread(new Runnable() {
           @Override
           public void run() {
-            reactContext.getCurrentActivity().getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE);
+            activity.getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE,
+                WindowManager.LayoutParams.FLAG_SECURE);
           }
         });
       }
     }
   }
-  
+
   @ReactMethod
   public void disableSecureView() {
     if (this.reactContext.hasCurrentActivity()) {
-      final Activity activity = this.reactContext.getCurrentActivity();
+      final Activity activity = this.activity;
       if (activity != null) {
         activity.runOnUiThread(new Runnable() {
           @Override
           public void run() {
-            reactContext.getCurrentActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
+            activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
           }
         });
       }
